snapshots/snapshot_generator/src/main/scala/uk/ac/wellcome/platform/snapshot_generator/akka/sink/CountingSink.scala

```diff
@@ -0,0 +1,11 @@
+package uk.ac.wellcome.platform.snapshot_generator.akka.sink
+
+import akka.stream.scaladsl.Sink
+import uk.ac.wellcome.display.models.DisplayWork
+
+import scala.concurrent.Future
+
+object CountingSink {
+  def apply(): Sink[DisplayWork, Future[Int]] =
+    Sink.fold[Int, DisplayWork](zero = 0)((total, _) => total + 1)
```

Comment:
Nothing about this is DisplayWork-specific, consider:

```suggestion
  def apply[T](): Sink[T, Future[Int]] =
    Sink.fold[Int, T](zero = 0)((total, _) => total + 1)
```